src/LocalDatastoreController.js

```diff
@@ -8,40 +8,40 @@
  *
  * @flow
  */
-
-/* global localStorage */
 import { isLocalDatastoreKey } from './LocalDatastoreUtils';
+import Storage from './Storage';
 
 const LocalDatastoreController = {
-  fromPinWithName(name: string): Array<Object> {
-    const values = localStorage.getItem(name);
+  async fromPinWithName(name: string): Array<Object> {
+    const values = await Storage.getItemAsync(name);
     if (!values) {
       return [];
     }
     const objects = JSON.parse(values);
     return objects;
   },
 
-  pinWithName(name: string, value: any) {
+  async pinWithName(name: string, value: any) {
     try {
       const values = JSON.stringify(value);
-      localStorage.setItem(name, values);
+      await Storage.setItemAsync(name, values);
     } catch (e) {
       // Quota exceeded, possibly due to Safari Private Browsing mode
       console.log(e.message);
     }
   },
 
   unPinWithName(name: string) {
-    localStorage.removeItem(name);
+    return Storage.removeItemAsync(name);
   },
 
-  getAllContents(): Object {
+  async getAllContents(): Object {
+    const keys = await Storage.getAllKeysAsync();
     const LDS = {};
-    for (let i = 0; i < localStorage.length; i += 1) {
-      const key = localStorage.key(i);
+
+    for(const key of keys){
```

Comment:
this is fine, but this kind of pattern I would use reduce.

Reply:
Please help great one. I tried using reduce but gave up.

Reply:
sorry, i'm wrong, localstorage is not iterable, so you can't use map and reduce on it...

Reply:
I'm looping through a set of keys and building a json object. localStorage isn't there anymore.

Reply:
right, one sec.

Reply:
https://runkit.com/arthur/reduce

Reply:
What about the async / await?

Reply:
make the reducer function async:

```js
array.reduce(async(accumulator...
```

Reply:
That would be cool, I also figured out how to use reduce, I forgot to return